python/cost_models.py

```diff
@@ -0,0 +1,220 @@
+import argparse
+import joblib
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+import pandas as pd
+import warnings
+
+from prepare_calibration_data import import_train_data
+from prepare_tpch import import_test_data
+from sklearn.ensemble import GradientBoostingRegressor
+from sklearn.linear_model import Lasso
+from sklearn.linear_model import LinearRegression
+from sklearn.linear_model import Ridge
+from sklearn.metrics import mean_squared_error
+from sklearn.model_selection import train_test_split
+plt.style.use('ggplot')
+
+def preprocess_data(data):
+    # one-hot encoding
+    ohe_data = data.drop(labels=['TABLE_NAME', 'COLUMN_NAME'], axis=1)
+    ohe_data = pd.get_dummies(ohe_data, columns=['SCAN_TYPE', 'DATA_TYPE', 'ENCODING', 'SCAN_IMPLEMENTATION', 'COMPRESSION_TYPE'])
+    return ohe_data
+
+
+def train_model(train_data, type):
+    ohe_data = preprocess_data(train_data)
+    y = np.ravel(ohe_data[['RUNTIME_NS']])
+    X = ohe_data.drop(labels=['RUNTIME_NS'], axis=1)
+    if type == 'linear':
+        model = LinearRegression().fit(X, y)
+    elif type == 'ridge':
+        model = Ridge(alpha=1000).fit(X, y)
+    elif type == 'lasso':
+        model = Lasso(alpha=1000).fit(X, y)
+    elif type == 'boost':
+        model = GradientBoostingRegressor(loss='huber').fit(X, y)
+
+    return model
+
+
+def generate_model_plot(model, test_data, method, encoding, scan, out):
+    ohe_data = preprocess_data(test_data)
+    real_y = np.ravel(ohe_data[['RUNTIME_NS']])
+    ohe_data = ohe_data.drop(labels=['RUNTIME_NS'], axis=1)
+    pred_y = model.predict(ohe_data)
+
+    model_scores = calculate_error(ohe_data, pred_y, real_y, model)
+
+    plt.scatter(real_y, pred_y, c='orange')
+
+    pred_y = model.predict(ohe_data)
+    axis_max = max(np.amax(pred_y), np.amax(real_y)) * 1.05
+    axis_min = min(np.amin(pred_y), np.amin(real_y)) * 0.95
+    abline_values = range(int(axis_min), int(axis_max), int((axis_max-axis_min)/100))
+
+    # Plot the best fit line over the actual values
+    plt.plot(abline_values, abline_values, c = 'r', linestyle="-")
+    plt.title('{}_{}_{}; Score: {}'.format(encoding, scan, method, model_scores['R2']))
+    plt.ylim([axis_min, axis_max])
+    plt.xlim([axis_min, axis_max])
+    plt.xlabel("Real Time")
+    plt.ylabel("Predicted Time")
+    output_path = '{}/Plots/{}_{}_{}'.format(out, method, encoding, scan)
```

Comment:
For the future, consider using python's f-strings.

Reply:
As it is a quick change and we already use it in pretty much all of our python scripts, I'd say we should switch here as well.

Reply:
Changed to f-strings